src/xGadgetron/cGadgetron/encoding.cpp

```diff
@@ -0,0 +1,311 @@
+#include "sirf/Gadgetron/encoding.h"
+
+#include <sstream>
+#include <math.h>
+
+//#include <cassert>
+//#define assertm(exp, msg) assert(((void)msg, exp)
+
+#define SIRF_GOLDEN_ANGLE M_PI*0.618034
```

Comment:
```suggestion
#define constexpr double SIRF_GOLDEN_ANGLE  = M_PI*0.618034
```
However, I'm fairly sure that `M_PI` isn't always going to be there, including on Windows. Might need its own constant.